src/gwt/src/org/rstudio/core/client/cellview/ImageButtonColumn.java

```diff
@@ -14,52 +14,124 @@
  */
 package org.rstudio.core.client.cellview;
 
+import static com.google.gwt.dom.client.BrowserEvents.CLICK;
+import static com.google.gwt.dom.client.BrowserEvents.KEYDOWN;
+
 import org.rstudio.core.client.resources.ImageResource2x;
 import org.rstudio.core.client.widget.OperationWithInput;
 
-import com.google.gwt.cell.client.ButtonCell;
+import com.google.gwt.cell.client.AbstractCell;
 import com.google.gwt.cell.client.FieldUpdater;
-import com.google.gwt.safehtml.shared.SafeHtml;
+import com.google.gwt.cell.client.ValueUpdater;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.EventTarget;
+import com.google.gwt.dom.client.NativeEvent;
 import com.google.gwt.safehtml.shared.SafeHtmlBuilder;
 import com.google.gwt.user.cellview.client.Column;
 
-public class ImageButtonColumn<T> extends Column<T, String>
+public class ImageButtonColumn<T> extends Column<T, T>
 {
+   public static interface TitleProvider<U>
+   {
+      public String get(U object);
+   }
+   
+   private static class ImageButtonCell<U> extends AbstractCell<U>
+   {
+      public ImageButtonCell(final ImageResource2x image,
+                             final TitleProvider<U> titleProvider)
+      {
+         super(CLICK, KEYDOWN);
+         
+         image_ = image;
+         titleProvider_ = titleProvider;
+      }
+      
+      @Override
+      public void render(Context context,
+                         U value,
+                         SafeHtmlBuilder sb)
+      {
+         if (value != null)
+         {
+            sb.appendHtmlConstant("<span title=\"" + titleProvider_.get(value) + "\" " +
```

Comment:
The title should be HTML escaped here if it's going into an HTML constant (which is presumed safe)